### PR TITLE
Allow any GenServer to act as a consumer managed by ConsumerGroup.

### PR DESCRIPTION
**Pull request by gerbal**
This PR allows any module implementing GenServer to used in place of KafkaEx.GenConsumer. 

```elixir
defmodule MyApp do
  use Application

  def start(_type, _args) do
    import Supervisor.Spec

    consumer_group_opts = [
      # setting for the ConsumerGroup
      heartbeat_interval: 1_000,
      # this setting will be forwarded to the GenConsumer
      commit_interval: 1_000,
      extra_consumer_args: [],
      commit_strategy: :async_commit
    ]

    consumer_group_name = "example_group"
    topic_names = ["example_topic"]

    children = [
      # ... other children
      supervisor(
        KafkaEx.ConsumerGroup,
        [{AlternativeGenericGenConsumer, MyApp.AnGenConsumer}, consumer_group_name, topic_names, consumer_group_opts]
      )
    ]

    Supervisor.start_link(children, strategy: :one_for_one)
  end
end
```

My primary motivation for adding this is to support [kafka_ex_gen_stage_consumer](https://github.com/sevenmind/kafka_ex_gen_stage_consumer), which is a GenStage producer that can be used to consume events with backpressure (Kind of addressing #236).

### CODE REVIEW

File: lib/kafka_ex/gen_consumer.ex [hunk 1]
```diff
@@ -175,7 +175,7 @@ defmodule KafkaEx.GenConsumer do
     @partition 0
 
     setup do
-      {:ok, state} = ExampleGenConsumer.init(@topic, @partition)
+      {k:ok, state} = ExampleGenConsumer.init(@topic, @partition)
```
Comment (joshuawscott) [inline lib/kafka_ex/gen_consumer.ex]: typo here?

Comment (gerbal) [reply lib/kafka_ex/gen_consumer.ex]: oops, fixed

**Comment by gerbal** [comment]
@joshuawscott How's it looking?

File: lib/kafka_ex/consumer_group.ex [hunk 2]
```diff
@@ -135,14 +135,33 @@ defmodule KafkaEx.ConsumerGroup do
...
-  @spec start_link(module, binary, [binary], options) :: Supervisor.on_start()
-  def start_link(consumer_module, group_name, topics, opts \\ []) do
+  @spec start_link(module | {module, module}, binary, [binary], options) :: Supervisor.on_start()
+  def start_link(consumer_module, group_name, topics, opts \\ [])
+  def start_link(consumer_module, group_name, topics, opts)
+      when is_atom(consumer_module) do
+    start_link({KafkaEx.GenConsumer, consumer_module}, group_name, topics, opts)
+  end
+
+  @doc """
```
Comment (joshuawscott) [inline lib/kafka_ex/consumer_group.ex]: I think this `@doc` needs to be combined with the other function head's `@doc` for the documentation to be generated correctly?

**Pull request merged**